### PR TITLE
feat(alerts): New issue alert styles, issue filters ui

### PR DESCRIPTION
**Pull request by scttcper**
- New alert filters behind feature flag
- Redesigned issue "Alert Conditions" section
- Alert issue builder fields have been re-arranged (name now on the top)

closes https://getsentry.atlassian.net/browse/WOR-114
depends on #20412

old:
old

new:
screenshot

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/index.tsx
Comment (davidenwang) [inline src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/index.tsx:96]: I guess it doesn't really make any difference but `filters: IssueAlertConditionTemplate[]` might make more sense as filters more closely resemble conditions rather than actions (even though the types are identical)

File: src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/index.tsx
Comment (davidenwang) [inline src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/index.tsx:455]: nice new errors 👍

File: src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/index.tsx
Comment (davidenwang) [inline src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/index.tsx:410]: I know this isn't your code, but I don't think we need to destructure actionMatch out of the rule object since it just gets added back in to initialData at line:428 and I don't see any other place where actionMatch is used.

File: src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/ruleNode.tsx
Comment (davidenwang) [inline src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/ruleNode.tsx:269]: do we no longer need this warning message?

File: src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/index.tsx
Comment (davidenwang) [inline src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/index.tsx:540]: general design nit: I know this was just carried over from the old form, but some input fields seem to be a lot wider than necessary. For example, the value selector here which is wider than the other two:
image

maybe we can fix some of the selectors in a follow up PR

**Review by davidenwang — COMMENTED** [review]
🥇 thanks for picking this up! Left a few comments, should be good to go after I land the backend bit.

File: src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/ruleNode.tsx
Comment (scttcper) [reply src/sentry/static/sentry/app/views/settings/projectAlerts/issueEditor/ruleNode.tsx:269]: @robinrendle said "i dare you to remove that useless warning message"

**Comment by scttcper** [comment]
Shrunk number inputs a bit

image

**Review by davidenwang — APPROVED** [review]

**Pull request merged**
